modules/dnn/src/layers/fast_convolution/depthwise_convolution.cpp

```diff
@@ -0,0 +1,390 @@
+    int *ofstab = ofstab_.data();
+    int *yxtab = ofstab + padded_ksize;
+
+    for (int k = 0; k < padded_ksize; k++)
+    {
+        int y = k < ksize ? k / Wk : 0;
+        int x = k < ksize ? k % Wk : 0;
+        int dy = y * dilation_y, dx = x * dilation_x;
+        yxtab[k * 2] = dy;
+        yxtab[k * 2 + 1] = dx;
+        ofstab[k] = dy * Wi + dx;
+    }
+
+    const float *weights0 = conv->weightsPtr, *bias = conv->biasPtr;
+    int inner_ytop = (pad_bottom + stride_y - 1) / stride_y, inner_ybottom = 3;
+    int inner_xleft = (pad_left + stride_x - 1) / stride_x, inner_xright = 4;
+
+    CV_Assert(ksize > 1 || (pad_left == 0 && pad_right == 0 && pad_top == 0 && pad_bottom == 0));
+
+    inner_xright = (Wi - (Wk - 1) * dilation_x + pad_left) / stride_x;
+    inner_xright += inner_xright * stride_x - pad_left + (Wk - 1) * dilation_x < Wi;
+    inner_ybottom = (Hi - (Hk - 1) * dilation_y + pad_top) / stride_y;
+    inner_ybottom += inner_ybottom * stride_y - pad_top + (Hk - 1) * dilation_y < Hi;
+
+    if (inner_xleft >= inner_xright || inner_ytop >= inner_ybottom)
+    {
+        inner_xleft = W0;
+        inner_ytop = H0;
+    }
+
+    inner_ybottom = inner_ybottom < H0 ? inner_ybottom : H0;
+
+    bool useSIMD = stride_x == 1 && inner_xleft < W0;
+    bool is3x3 = Hk == 3 && Wk == 3;
+
+    parallel_for_(Range(0, N * C), [&](const Range &r0) {
+        for (int nc = r0.start; nc < r0.end; nc++)
+        {
+            int c = nc % C;
+            const float *inptr = inp + inp_planesize * nc;
+            float *outptr0 = out + out_planesize * nc;
+
+            float biasval = bias[c];
+            const float *weights = weights0 + c * padded_ksize;
+
+#if CV_TRY_AVX2
+            if (conv->useAVX2)
+                opt_AVX2::depthWiseBlock(inptr, outptr0, weights, biasval, ofstab, yxtab, minval, maxval, Hi, Wi, H0, W0, ksize,
+                                         pad_top, pad_left, dilation_y, stride_x, stride_y, inner_xleft, inner_xright, inner_ytop,
+                                         inner_ybottom, ifMinMaxAct, useSIMD, is3x3);
+            else
+#endif
+#if CV_TRY_AVX
+            if (conv->useAVX)
+                opt_AVX::depthWiseBlock(inptr, outptr0, weights, biasval, ofstab, yxtab, minval, maxval, Hi, Wi, H0, W0, ksize,
+                                        pad_top, pad_left, dilation_y, stride_x, stride_y, inner_xleft, inner_xright, inner_ytop,
+                                        inner_ybottom, ifMinMaxAct, useSIMD, is3x3);
+            else
+#endif
+            depthWiseBlockSIMD(inptr, outptr0, weights, biasval, ofstab, yxtab, minval, maxval, Hi, Wi, H0, W0, ksize,
+                           pad_top, pad_left, dilation_y, stride_x, stride_y, inner_xleft, inner_xright, inner_ytop,
+                           inner_ybottom, ifMinMaxAct, useSIMD, is3x3);
+
+            if (activ)
+                activ->forwardSlice(outptr0, outptr0, (int) out_planesize, out_planesize, c, c+1);
```

Comment:
can you please make sure that in the case of ReLU/ReLU6(Clip) a separate activation function is not called twice? I do not see where it's handled

Reply:
I'm sure it can handle ReLU/ReLU6(Clip) correctly. The process of activation is at the beginning of `fastConv2d()`.